### ISSUE TITLE
UpdateDone leaks after the update to Akka Version 2.6.17 to 2.6.18

### ISSUE THREAD

**Issue by mehmetsalgar** [akka/akka#31100]
I previously reported a similar Issue #31091, which is classified as Bug and fixed while ShardsUpdated Event leaked to MessageExtractor.

Now during my regression tests, another version submerged, UpdateDone is also leaking to MessageExtractor.

[PR for 31091](https://github.com/johanandren/akka/commit/ff4295ddd96908f781966c58d6fa5cca13451348) introduces a new case statement for 'ShardUpdated' in Shard Class but UpdateDone is still not handled, I don't know this is the correct fix but my 2 cents.

java.lang.ClassCastException: class akka.cluster.sharding.internal.RememberEntitiesShardStore$UpdateDone cannot be cast to class org.example.TestActor$TestActorEvent (akka.cluster.sharding.internal.RememberEntitiesShardStore$UpdateDone and org.example.TestActor$TestEvent are in unnamed module of loader 'app')
	at org.example.TestActorGuardian$$anon$1.entityId(CreditScoreSMGuardian.scala:197)
	at akka.cluster.sharding.typed.internal.ExtractorAdapter.entityId(ClusterShardingImpl.scala:58)
	at akka.cluster.sharding.typed.internal.ClusterShardingImpl$$anonfun$1.isDefinedAt(ClusterShardingImpl.scala:168)
	at akka.cluster.sharding.Shard$$anonfun$idle$1.applyOrElse(Shard.scala:626)

**Comment by patriknw** [comment]
@pvlugter Can you take a look at this?

**Comment by pvlugter** [comment]
`ShardsUpdated` is a new message for passivation strategies. The `UpdateDone` message is for the remember entities feature, and it doesn't look like anything has changed there recently. The same issue has also been seen in tests before (for example, #30641).

We could add a case for `UpdateDone` to the `idle` receive, but does this actually point to some underlying issue in the state machine for remember entities?

**Comment by pvlugter** [comment]
Hi @mehmetsalgar, would you be able to reproduce this with debug level logging enable, and also enable this config setting:

```
akka.cluster.sharding.verbose-debug-logging = on
```

The debug logs should indicate what entities are being remembered and the updates. I can see the pattern that creates something similar in #30641, and want to check what you're seeing here as well.

**Comment by mehmetsalgar** [comment]
Ok,

with the parameters you mentioned, I reproduced the logs and stack traces....

I have to say, you might see that Remember Entities didn't received any changes but I when I downgrade to 2.6.17, I can't reproduce these errors, so there must be a connection. I don't quite get a passivation strategy has an effect for Remember Entities while if you look to stack traces bother Exceptions originate from same component..

ShardsUpdated -> at akka.cluster.sharding.Shard$$anonfun$waitingForRememberEntitiesStore$1.applyOrElse(Shard.scala:703)

UpdateDone -> java.lang.ClassCastException: class akka.cluster.sharding.internal.RememberEntitiesShardStore

------------------------------------------------------------------------------------------
[2022-02-02 11:16:33,218] [INFO ] [a.c.s.t.s.ClusterSharding] [test-akka.actor.default-dispatcher-3] [ClusterSharding(akka://test)] - Starting Shard Region [TestActor]...
[2022-02-02 11:16:33,225] [INFO ] [a.c.s.ClusterSingletonManager] [test-akka.actor.default-dispatcher-39] [akka://test@127.0.0.1:2551/system/sharding/TestActorCoordinator] - Singleton manager starting singleton actor [akka://test/system/sharding/TestActorCoordinator/singleton]
[2022-02-02 11:16:33,225] [INFO ] [a.c.s.ClusterSingletonManager] [test-akka.actor.default-dispatcher-39] [akka://test@127.0.0.1:2551/system/sharding/TestActorCoordinator] - ClusterSingletonManager state change [Start -> Oldest]
[2022-02-02 11:16:33,501] [DEBUG] [a.p.t.i.EventSourcedBehaviorImpl] [test-akka.actor.default-dispatcher-3] [akka://test/system/sharding/TestActor/12/0e679b7f-806f-41db-bd8e-21f3f62aa3e3_02_02_2022_123456789X] - Initializing snapshot recovery: Recovery(SnapshotSelectionCriteria(9223372036854775807,9223372036854775807,0,0),9223372036854775807,9223372036854775807)
[2022-02-02 11:16:33,503] [DEBUG] [a.p.t.i.EventSourcedBehaviorImpl] [test-akka.actor.default-dispatcher-39] [akka://test/system/sharding/TestActor/12/ce424b83-48a4-418e-b58f-37a549ca1656_123456789X] - Initializing snapshot recovery: Recovery(SnapshotSelectionCriteria(9223372036854775807,9223372036854775807,0,0),9223372036854775807,9223372036854775807)
[2022-02-02 11:16:33,503] [DEBUG] [a.p.t.i.EventSourcedBehaviorImpl] [test-akka.actor.default-dispatcher-39] [akka://test/system/sharding/TestActor/19/d084a5d4-21b4-4b33-92fc-43cd2fb7c7eb_123456789X] - Initializing snapshot recovery: Recovery(SnapshotSelectionCriteria(9223372036854775807,9223372036854775807,0,0),9223372036854775807,9223372036854775807)
[2022-02-02 11:16:33,504] [DEBUG] [a.p.t.i.EventSourcedBehaviorImpl] [test-akka.actor.default-dispatcher-3] [akka://test/system/sharding/TestActor/21/773fbe33-c039-4c5d-b0e3-9e8341111157_123456789X] - Initializing snapshot recovery: Recovery(SnapshotSelectionCriteria(9223372036854775807,9223372036854775807,0,0),9223372036854775807,9223372036854775807)
[2022-02-02 11:16:33,506] [DEBUG] [a.p.t.i.EventSourcedBehaviorImpl] [test-akka.actor.default-dispatcher-15] [akka://test/system/sharding/TestActor/23/0e679b7f-806f-41db-bd8e-21f3f62aa3e4_02_02_2022_Z987654321] - Initializing snapshot recovery: Recovery(SnapshotSelectionCriteria(9223372036854775807,9223372036854775807,0,0),9223372036854775807,9223372036854775807)
[2022-02-02 11:16:33,512] [ERROR] [a.a.OneForOneStrategy] [test-akka.actor.default-dispatcher-5] [akka://test/system/sharding/TestActor/26] - class akka.cluster.sharding.ShardRegion$ShardsUpdated cannot be cast to class org.example.TestActor$TestActorEvent (akka.cluster.sharding.ShardRegion$ShardsUpdated and org.example.TestActor$TestActorEvent are in unnamed module of loader 'app')
java.lang.ClassCastException: class akka.cluster.sharding.ShardRegion$ShardsUpdated cannot be cast to class org.example.TestActor$TestActorSMEvent (akka.cluster.sharding.ShardRegion$ShardsUpdated and org.example.TestActor$TestActorEvent are in unnamed module of loader 'app')
	at org.example.TestActorGuardian$$anon$1.entityId(TestActorGuardian.scala:197)
	at akka.cluster.sharding.typed.internal.ExtractorAdapter.entityId(ClusterShardingImpl.scala:58)
	at akka.cluster.sharding.typed.internal.ClusterShardingImpl$$anonfun$1.isDefinedAt(ClusterShardingImpl.scala:168)
	at akka.cluster.sharding.Shard$$anonfun$waitingForRememberEntitiesStore$1.applyOrElse(Shard.scala:703)
	at akka.actor.Actor.aroundReceive(Actor.scala:537)
	at akka.actor.Actor.aroundReceive$(Actor.scala:535)
	at akka.cluster.sharding.Shard.akka$actor$Timers$$super$aroundReceive(Shard.scala:410)
	at akka.actor.Timers.aroundReceive(Timers.scala:56)
	at akka.actor.Timers.aroundReceive$(Timers.scala:41)
	at akka.cluster.sharding.Shard.aroundReceive(Shard.scala:410)
	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:580)
	at akka.actor.ActorCell.invoke(ActorCell.scala:548)
	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:270)
	at akka.dispatch.Mailbox.run(Mailbox.scala:231)
	at akka.dispatch.Mailbox.exec(Mailbox.scala:243)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
[2022-02-02 11:16:33,515] [ERROR] [a.a.OneForOneStrategy] [test-akka.actor.default-dispatcher-18] [akka://test/system/sharding/TestActor/26] - class akka.cluster.sharding.internal.RememberEntitiesShardStore$UpdateDone cannot be cast to class org.example.TestActor$TestActorEvent (akka.cluster.sharding.internal.RememberEntitiesShardStore$UpdateDone and org.example.TestActor$TestActorEvent are in unnamed module of loader 'app')
java.lang.ClassCastException: class akka.cluster.sharding.internal.RememberEntitiesShardStore$UpdateDone cannot be cast to class org.example.TestActor$TestActorEvent (akka.cluster.sharding.internal.RememberEntitiesShardStore$UpdateDone and org.example.TestActor$TestActorEvent are in unnamed module of loader 'app')
	at org.example.TestActorGuardian$$anon$1.entityId(TestActorGuardian.scala:197)
	at akka.cluster.sharding.typed.internal.ExtractorAdapter.entityId(ClusterShardingImpl.scala:58)
	at akka.cluster.sharding.typed.internal.ClusterShardingImpl$$anonfun$1.isDefinedAt(ClusterShardingImpl.scala:168)
	at akka.cluster.sharding.Shard$$anonfun$idle$1.applyOrElse(Shard.scala:626)
	at akka.actor.Actor.aroundReceive(Actor.scala:537)
	at akka.actor.Actor.aroundReceive$(Actor.scala:535)
	at akka.cluster.sharding.Shard.akka$actor$Timers$$super$aroundReceive(Shard.scala:410)
	at akka.actor.Timers.aroundReceive(Timers.scala:56)
	at akka.actor.Timers.aroundReceive$(Timers.scala:41)
	at akka.cluster.sharding.Shard.aroundReceive(Shard.scala:410)
	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:580)
	at akka.actor.ActorCell.invoke(ActorCell.scala:548)
	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:270)
	at akka.dispatch.Mailbox.run(Mailbox.scala:231)

**Comment by pvlugter** [comment]
Thanks @mehmetsalgar. The `ShardsUpdated` message being unhandled and crashing the actor could certainly be the problem, putting the actor back into the idle state when the `UpdateDone` messages are received. I had assumed this was a new issue and was already running with the fix for `ShardsUpdated`.

It would be useful to confirm that things do work for you with the recent fix. You can use a published Akka snapshot version. You'll need to include the Sonatype snapshots repository: https://oss.sonatype.org/content/repositories/snapshots

So in sbt, that would be including:

```
resolvers += "sonatype snapshots" at "https://oss.sonatype.org/content/repositories/snapshots"
```

And then use the latest snapshot for the Akka version: `2.6.18+46-f16051c0-SNAPSHOT`

**Comment by mehmetsalgar** [comment]
I am trying to run my regression tests with previous PRs fixes in this snapshot version but Akka Logging System is not initialising with this snapshot version (may be because of a log4j vulnerability fix?) with the following exception.

If I downgrade from snapshot to 2.6.18, I don't get this logging error.

Any idea how to fix this logging error?

-------
Caused by: akka.ConfigurationException: Could not start logger due to [akka.ConfigurationException: Logger specified in config can't be loaded [akka.event.slf4j.Slf4jLogger] due to [akka.event.Logging$LoggerInitializationException: Logger log1-Slf4jLogger did not respond with LoggerInitialized, sent instead [TIMEOUT]]]
	at akka.event.LoggingBus.startDefaultLoggers(Logging.scala:165)
	at akka.event.LoggingBus.startDefaultLoggers$(Logging.scala:108)
	at akka.event.EventStream.startDefaultLoggers(EventStream.scala:26)
	at akka.actor.LocalActorRefProvider.init(ActorRefProvider.scala:609)
	at akka.remote.RemoteActorRefProvider.init(RemoteActorRefProvider.scala:221)
	at akka.cluster.ClusterActorRefProvider.init(ClusterActorRefProvider.scala:49)

**Comment by johanandren** [comment]
That is curious, but you could try increasing the default logger init timeout (5s) with the setting `akka.logger-startup-timeout`

**Comment by mehmetsalgar** [comment]
I tried with 

logger-startup-timeout = 120s

still the same problem

**Comment by johanandren** [comment]
Must be some form of deadlock with logging then, I remember I have seen a report of that a couple of years ago because some issue with the logging setup, but I wonder if that wasn't some JDK issue. Can you try touching slf4j directly, logging something in your `main` before even starting the actor system and see if that sorts it out?

**Comment by mehmetsalgar** [comment]
I followed you advice to log something before starting anything in my application, still. no success...

Funny thing, if I go from this SNAPSHOT version back to 2.18,  Logging System is initialising without any problem, my code is exactly the same.

Something in the Logging System must be different in Akka between 2.18 and 2.18-SNAPSHOT.

**Comment by patriknw** [comment]
Thanks for your attempts. Doesn't sound good if something messed up with the logging. Can you provide the exact dependencies that you use (akka snapshot, logback, slf4j) so that we can try that in a separate example project.

**Comment by mehmetsalgar** [comment]
These are the logback libraries that I see

|    |    +--- ch.qos.logback:logback-classic:1.2.10
|    |    |    +--- ch.qos.logback:logback-core:1.2.10
|    |    |    \--- org.slf4j:slf4j-api:1.7.32 -> 1.7.33
|    |    +--- org.apache.logging.log4j:log4j-to-slf4j:2.17.1
|    |    |    +--- org.slf4j:slf4j-api:1.7.25 -> 1.7.33
|    |    |    \--- org.apache.logging.log4j:log4j-api:2.17.1
|    |    \--- org.slf4j:jul-to-slf4j:1.7.33
|    |         \--- org.slf4j:slf4j-api:1.7.33

And Akka Dependencies
|    |    |    +--- com.typesafe.akka:akka-actor-typed_2.13:2.6.18+46-f16051c0-SNAPSHOT (*)
|    |    |    +--- com.typesafe.akka:akka-persistence-typed_2.13:2.6.18+46-f16051c0-SNAPSHOT (*)
|    |    |    +--- com.typesafe.akka:akka-slf4j_2.13:2.6.18+46-f16051c0-SNAPSHOT (*)
|    |    |    +--- com.typesafe.akka:akka-cluster-sharding-typed_2.13:2.6.18+46-f16051c0-SNAPSHOT
|    |    |    |    +--- org.scala-lang:scala-library:2.13.7
|    |    |    |    +--- com.typesafe.akka:akka-actor-typed_2.13:2.6.18+46-f16051c0-SNAPSHOT (*)
|    |    |    |    +--- com.typesafe.akka:akka-cluster-typed_2.13:2.6.18+46-f16051c0-SNAPSHOT
|    |    |    |    |    +--- org.scala-lang:scala-library:2.13.7
|    |    |    |    |    +--- com.typesafe.akka:akka-actor-typed_2.13:2.6.18+46-f16051c0-SNAPSHOT (*)
|    |    |    |    |    +--- com.typesafe.akka:akka-cluster_2.13:2.6.18+46-f16051c0-SNAPSHOT
|    |    |    |    |    |    +--- org.scala-lang:scala-library:2.13.7
|    |    |    |    |    |    +--- com.typesafe.akka:akka-remote_2.13:2.6.18+46-f16051c0-SNAPSHOT (*)
|    |    |    |    |    |    \--- com.typesafe.akka:akka-coordination_2.13:2.6.18+46-f16051c0-SNAPSHOT
|    |    |    |    |    |         +--- org.scala-lang:scala-library:2.13.7
|    |    |    |    |    |         \--- com.typesafe.akka:akka-actor_2.13:2.6.18+46-f16051c0-SNAPSHOT (*)
|    |    |    |    |    +--- com.typesafe.akka:akka-cluster-tools_2.13:2.6.18+46-f16051c0-SNAPSHOT
|    |    |    |    |    |    +--- org.scala-lang:scala-library:2.13.7
|    |    |    |    |    |    +--- com.typesafe.akka:akka-cluster_2.13:2.6.18+46-f16051c0-SNAPSHOT (*)
|    |    |    |    |    |    \--- com.typesafe.akka:akka-coordination_2.13:2.6.18+46-f16051c0-SNAPSHOT (*)
|    |    |    |    |    \--- com.typesafe.akka:akka-distributed-data_2.13:2.6.18+46-f16051c0-SNAPSHOT
|    |    |    |    |         +--- org.scala-lang:scala-library:2.13.7
|    |    |    |    |         +--- com.typesafe.akka:akka-cluster_2.13:2.6.18+46-f16051c0-SNAPSHOT (*)
|    |    |    |    |         \--- org.lmdbjava:lmdbjava:0.7.0
|    |    |    |    |              +--- com.github.jnr:jnr-ffi:2.1.9 -> 2.2.2
|    |    |    |    |              |    +--- com.github.jnr:jffi:1.3.1
|    |    |    |    |              |    +--- org.ow2.asm:asm:9.1
|    |    |    |    |              |    +--- org.ow2.asm:asm-commons:9.1
|    |    |    |    |              |    |    +--- org.ow2.asm:asm:9.1
|    |    |    |    |              |    |    +--- org.ow2.asm:asm-tree:9.1
|    |    |    |    |              |    |    |    \--- org.ow2.asm:asm:9.1
|    |    |    |    |              |    |    \--- org.ow2.asm:asm-analysis:9.1
|    |    |    |    |              |    |         \--- org.ow2.asm:asm-tree:9.1 (*)
|    |    |    |    |              |    +--- org.ow2.asm:asm-analysis:9.1 (*)
|    |    |    |    |              |    +--- org.ow2.asm:asm-tree:9.1 (*)
|    |    |    |    |              |    +--- org.ow2.asm:asm-util:9.1
|    |    |    |    |              |    |    +--- org.ow2.asm:asm:9.1
|    |    |    |    |              |    |    +--- org.ow2.asm:asm-tree:9.1 (*)
|    |    |    |    |              |    |    \--- org.ow2.asm:asm-analysis:9.1 (*)
|    |    |    |    |              |    +--- com.github.jnr:jnr-a64asm:1.0.0
|    |    |    |    |              |    \--- com.github.jnr:jnr-x86asm:1.0.2
|    |    |    |    |              +--- com.github.jnr:jffi:1.2.18 -> 1.3.1
|    |    |    |    |              \--- com.github.jnr:jnr-constants:0.9.12 -> 0.10.1
|    |    |    |    +--- com.typesafe.akka:akka-cluster-sharding_2.13:2.6.18+46-f16051c0-SNAPSHOT
|    |    |    |    |    +--- org.scala-lang:scala-library:2.13.7
|    |    |    |    |    +--- com.typesafe.akka:akka-cluster_2.13:2.6.18+46-f16051c0-SNAPSHOT (*)
|    |    |    |    |    +--- com.typesafe.akka:akka-distributed-data_2.13:2.6.18+46-f16051c0-SNAPSHOT (*)
|    |    |    |    |    +--- com.typesafe.akka:akka-persistence_2.13:2.6.18+46-f16051c0-SNAPSHOT (*)
|    |    |    |    |    \--- com.typesafe.akka:akka-cluster-tools_2.13:2.6.18+46-f16051c0-SNAPSHOT (*)
|    |    |    |    \--- com.typesafe.akka:akka-remote_2.13:2.6.18+46-f16051c0-SNAPSHOT (*)
|    |    |    +--- com.lightbend.akka.management:akka-management-cluster-http_2.13:1.1.1
|    |    |    |    +--- com.lightbend.akka.management:akka-management_2.13:1.1.1
|    |    |    |    |    +--- com.typesafe.akka:akka-actor_2.13:2.6.14 -> 2.6.18+46-f16051c0-SNAPSHOT (*)
|    |    |    |    |    +--- com.typesafe.akka:akka-stream_2.13:2.6.14 -> 2.6.18+46-f16051c0-SNAPSHOT (*)
|    |    |    |    |    +--- com.typesafe.akka:akka-http_2.13:10.2.0
|    |    |    |    |    |    +--- org.scala-lang:scala-library:2.13.3 -> 2.13.7
|    |    |    |    |    |    \--- com.typesafe.akka:akka-http-core_2.13:10.2.0
|    |    |    |    |    |         +--- org.scala-lang:scala-library:2.13.3 -> 2.13.7
|    |    |    |    |    |         \--- com.typesafe.akka:akka-parsing_2.13:10.2.0
|    |    |    |    |    |              \--- org.scala-lang:scala-library:2.13.3 -> 2.13.7
|    |    |    |    |    \--- com.typesafe.akka:akka-http-spray-json_2.13:10.2.0
|    |    |    |    |         +--- org.scala-lang:scala-library:2.13.3 -> 2.13.7
|    |    |    |    |         +--- com.typesafe.akka:akka-http_2.13:10.2.0 (*)
|    |    |    |    |         \--- io.spray:spray-json_2.13:1.3.5
|    |    |    |    |              \--- org.scala-lang:scala-library:2.13.0 -> 2.13.7
|    |    |    |    +--- com.typesafe.akka:akka-cluster_2.13:2.6.14 -> 2.6.18+46-f16051c0-SNAPSHOT (*)
|    |    |    |    +--- com.typesafe.akka:akka-cluster-sharding_2.13:2.6.14 -> 2.6.18+46-f16051c0-SNAPSHOT (*)
|    |    |    |    +--- com.typesafe.akka:akka-http-core_2.13:10.2.0 (*)
|    |    |    |    \--- com.typesafe.akka:akka-http-spray-json_2.13:10.2.0 (*)
|    |    |    +--- com.lightbend.akka.management:akka-management-cluster-bootstrap_2.13:1.1.1
|    |    |    |    +--- com.lightbend.akka.management:akka-management_2.13:1.1.1 (*)
|    |    |    |    +--- com.typesafe.akka:akka-discovery_2.13:2.6.14 -> 2.6.18+46-f16051c0-SNAPSHOT
|    |    |    |    |    +--- org.scala-lang:scala-library:2.13.7
|    |    |    |    |    \--- com.typesafe.akka:akka-actor_2.13:2.6.18+46-f16051c0-SNAPSHOT (*)
|    |    |    |    +--- com.typesafe.akka:akka-cluster_2.13:2.6.14 -> 2.6.18+46-f16051c0-SNAPSHOT (*)
|    |    |    |    +--- com.typesafe.akka:akka-http-core_2.13:10.2.0 (*)
|    |    |    |    \--- com.typesafe.akka:akka-http-spray-json_2.13:10.2.0 (*)
|    |    |    +--- com.lightbend.akka.discovery:akka-discovery-kubernetes-api_2.13:1.1.1
|    |    |    |    +--- com.lightbend.akka.management:akka-management-pki_2.13:1.1.1
|    |    |    |    |    \--- com.typesafe.akka:akka-pki_2.13:2.6.14 -> 2.6.18+46-f16051c0-SNAPSHOT (*)
|    |    |    |    +--- com.typesafe.akka:akka-actor_2.13:2.6.14 -> 2.6.18+46-f16051c0-SNAPSHOT (*)
|    |    |    |    +--- com.typesafe.akka:akka-discovery_2.13:2.6.14 -> 2.6.18+46-f16051c0-SNAPSHOT (*)
|    |    |    |    +--- com.typesafe.akka:akka-stream_2.13:2.6.14 -> 2.6.18+46-f16051c0-SNAPSHOT (*)
|    |    |    |    +--- com.typesafe.akka:akka-http_2.13:10.2.0 (*)
|    |    |    |    \--- com.typesafe.akka:akka-http-spray-json_2.13:10.2.0 (*)
|    |    |    +--- com.typesafe.akka:akka-discovery_2.13:2.6.18+46-f16051c0-SNAPSHOT (*)
|    |    +--- com.typesafe.akka:akka-cluster-sharding-typed_2.13:2.6.18+46-f16051c0-SNAPSHOT (*)
+--- com.typesafe.akka:akka-persistence-cassandra_2.13:1.0.5
|    +--- org.scala-lang:scala-library:2.13.3 -> 2.13.7
|    +--- com.lightbend.akka:akka-stream-alpakka-cassandra_2.13:2.0.2
|    |    +--- org.scala-lang:scala-library:2.13.3 -> 2.13.7
|    |    +--- com.typesafe.akka:akka-stream_2.13:2.5.31 -> 2.6.18+46-f16051c0-SNAPSHOT (*)
|    |    +--- com.datastax.oss:java-driver-core:4.6.1 -> 4.13.0
|    |    |    +--- com.datastax.oss:native-protocol:1.5.0
|    |    |    +--- io.netty:netty-handler:4.1.60.Final -> 4.1.73.Final
|    |    |    |    +--- io.netty:netty-common:4.1.73.Final
|    |    |    |    +--- io.netty:netty-resolver:4.1.73.Final
|    |    |    |    |    \--- io.netty:netty-common:4.1.73.Final
|    |    |    |    +--- io.netty:netty-buffer:4.1.73.Final
|    |    |    |    |    \--- io.netty:netty-common:4.1.73.Final
|    |    |    |    +--- io.netty:netty-transport:4.1.73.Final
|    |    |    |    |    +--- io.netty:netty-common:4.1.73.Final
|    |    |    |    |    +--- io.netty:netty-buffer:4.1.73.Final (*)
|    |    |    |    |    \--- io.netty:netty-resolver:4.1.73.Final (*)
|    |    |    |    +--- io.netty:netty-codec:4.1.73.Final
|    |    |    |    |    +--- io.netty:netty-common:4.1.73.Final
|    |    |    |    |    +--- io.netty:netty-buffer:4.1.73.Final (*)
|    |    |    |    |    \--- io.netty:netty-transport:4.1.73.Final (*)
|    |    |    |    \--- io.netty:netty-tcnative-classes:2.0.46.Final
|    |    |    +--- com.datastax.oss:java-driver-shaded-guava:25.1-jre-graal-sub-1
|    |    |    +--- com.typesafe:config:1.4.1
|    |    |    +--- com.github.jnr:jnr-posix:3.1.5
|    |    |    |    +--- com.github.jnr:jnr-ffi:2.2.2 (*)
|    |    |    |    \--- com.github.jnr:jnr-constants:0.10.1
|    |    |    +--- org.slf4j:slf4j-api:1.7.26 -> 1.7.33
|    |    |    +--- io.dropwizard.metrics:metrics-core:4.1.18 -> 4.2.7
|    |    |    |    \--- org.slf4j:slf4j-api:1.7.32 -> 1.7.33
|    |    |    +--- org.hdrhistogram:HdrHistogram:2.1.12
|    |    |    +--- com.fasterxml.jackson.core:jackson-core:2.12.2 -> 2.13.1 (*)
|    |    |    +--- com.fasterxml.jackson.core:jackson-databind:2.12.2 -> 2.13.1 (*)
|    |    |    +--- org.reactivestreams:reactive-streams:1.0.3
|    |    |    \--- com.github.stephenc.jcip:jcip-annotations:1.0-1
|    |    +--- io.netty:netty-handler:4.1.52.Final -> 4.1.73.Final (*)
|    |    +--- com.fasterxml.jackson.core:jackson-core:2.10.5 -> 2.13.1 (*)
|    |    \--- com.fasterxml.jackson.core:jackson-databind:2.10.5 -> 2.13.1 (*)
|    +--- com.typesafe.akka:akka-persistence_2.13:2.6.9 -> 2.6.18+46-f16051c0-SNAPSHOT (*)
|    +--- com.typesafe.akka:akka-persistence-query_2.13:2.6.9 -> 2.6.18+46-f16051c0-SNAPSHOT (*)
|    \--- com.typesafe.akka:akka-cluster-tools_2.13:2.6.9 -> 2.6.18+46-f16051c0-SNAPSHOT (*)
+--- com.typesafe.akka:akka-persistence-query_2.13:2.6.18+46-f16051c0-SNAPSHOT (*)
+--- com.typesafe.akka:akka-persistence-typed_2.13:2.6.18+46-f16051c0-SNAPSHOT (*)
+--- com.typesafe.akka:akka-cluster-sharding-typed_2.13:2.6.18+46-f16051c0-SNAPSHOT (*)

**Comment by pvlugter** [comment]
I tried reproducing the logging initialisation issue with the same versions, but no luck so far.

**Comment by patriknw** [comment]
I also tried `2.6.18+46-f16051c0-SNAPSHOT` with https://github.com/akka/akka-platform-guide/tree/main/docs-source/docs/modules/microservices-tutorial/examples/01-shopping-cart-service-scala

It starts up as expected.

added dependencies:
```
  "ch.qos.logback" % "logback-classic" % "1.2.10",
  "org.apache.logging.log4j" % "log4j-to-slf4j" "2.17.1",
  "org.slf4j" % "jul-to-slf4j" % "1.7.33",
```

**Comment by pvlugter** [comment]
@mehmetsalgar, being able to reproduce with your project would be useful. Please update here when it's available. Thanks!

**Comment by patriknw** [comment]
I assume this was "fixed" since we didn't hear anything more.
